packages/gatsby/src/redux/run-sift.js

```diff
@@ -166,12 +166,8 @@ const filterWithoutSift = (filters, nodeTypeNames, filtersCache) => {
       b.length - a.length
   )
 
-  if (nodesPerValueArrs.length === 1) {
```

Comment:
Note: this is the main step I'm trying to prevent because this happens at "runtime" (n times) not "cache generation time" (one time), even for single filters. It's just better on all accounts :)